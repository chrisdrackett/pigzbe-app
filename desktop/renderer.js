import {AppRegistry} from 'react-native';
import App from './bundle.web.js';

AppRegistry.registerComponent('App', () => App);
AppRegistry.runApplication('App', {rootTag: document.getElementById('react-root')});
