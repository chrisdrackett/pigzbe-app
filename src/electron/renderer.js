import {AppRegistry} from 'react-native';
// import App from '../../App';
import App from '../../dist/bundle.web.js';

AppRegistry.registerComponent('App', () => App);
AppRegistry.runApplication('App', {rootTag: document.getElementById('react-root')});
