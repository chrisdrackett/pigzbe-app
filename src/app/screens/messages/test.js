import React from 'react';
import renderer from 'react-test-renderer';
import {Messages} from './';

const props = {
    dispatch: () => {},
    navigation: {
        navigate: () => {},
        addListener: () => {},
        state: {
            key: 'SCREEN_MESSAGES',
            routeName: 'SCREEN_MESSAGES'
        },
        actions: {}
    },
    messages: [{
        key: '1',
        date: '2018-04-20T12:00+01:00',
        text: 'Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.',
        link: 'https://pigzbe.com'
    }, {
        key: '2',
        date: '2018-04-20T09:00+01:00',
        text: 'Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.'
    }],
    loading: false,
    error: null
};

describe('Messages', () => {
    test('renders correctly with messages', () => {
        renderer.create(
            <Messages {...props}/>
        );
    });

    test('renders correctly without messages', () => {
        renderer.create(
            <Messages {...Object.assign({}, props, {
                messages: []
            })}/>
        );
    });
});
