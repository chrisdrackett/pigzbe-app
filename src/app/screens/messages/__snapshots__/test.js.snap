// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages renders correctly without messages 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgb(0, 50, 120)",
        "flex": 1,
        "justifyContent": "space-between",
      },
      Object {
        "width": 750,
      },
      Object {
        "display": "flex",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(21, 74, 150)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 242,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 50, 120)",
        "justifyContent": "center",
        "minHeight": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/app/components/logo/images/pigzbe_logo_inapp.png",
        }
      }
      style={
        Object {
          "height": 25,
          "marginBottom": 10,
          "marginTop": 34,
          "width": 65,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(255, 255, 255)",
            "fontFamily": "Poppins",
            "fontSize": 22,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Your Updates
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/app/components/step-header/images/messages.png",
        }
      }
      style={
        Object {
          "height": 100,
          "marginBottom": -30,
          "marginTop": 8,
          "width": 200,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexGrow": 1,
          },
          Object {
            "bottom": 0,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgb(0, 50, 120)",
              "flex": 1,
              "justifyContent": "space-between",
            },
            Object {
              "width": 750,
            },
            Array [
              Object {
                "alignSelf": "center",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "width": "88.75%",
              },
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
              },
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flex": 1,
              },
              null,
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "alignSelf": "stretch",
                "position": "relative",
              }
            }
          >
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
